OOVPADatabase/XOnline/4831.inl

```diff
@@ -49,3 +49,205 @@ OOVPA_XREF(CXo_XOnlineLogon, 4831, 15,
+
+        // mov ecx, dword ptr [ebp + param_1]
+        // push ebx
+        OV_MATCH(0x11, 0x8B, 0x4D, 0x08, 0x53),
+OOVPA_END;
+
+// ******************************************************************
+// * XOnlineMatchSearchGetResults
+// ******************************************************************
+OOVPA_XREF(XOnlineMatchSearchGetResults, 4831, 1+3,
+
+    XRefNoSaveIndex,
+    XRefOne)
+
+        XREF_ENTRY(0x07, XREF_CXo_XOnlineMatchSearchGetResults),
+
+        // mov ecx, ...
+        OV_MATCH(0x00, 0x8B, 0x0D),
+
+        // jmp ...
+        OV_MATCH(0x06, 0xE9),
+OOVPA_END;
+
+// ******************************************************************
+// * CXo::XOnlineMatchSessionUpdate
+// ******************************************************************
+OOVPA_XREF(CXo_XOnlineMatchSessionUpdate, 4831, 22,
+
+    XREF_CXo_XOnlineMatchSessionUpdate,
+    XRefZero)
+
+        // push ebp
+        // mov ebp, esp
+        // test ecx, ecx
+        // jnz eip + $07h
+        OV_MATCH(0x00, 0x55, 0x8B, 0xEC, 0x85, 0xC9, 0x75, 0x07),
+
+        // push dword ptr [ebp + param_10]
+        // push dword ptr [ebp + param_9]
+        // push dword ptr [ebp + param_8]
+        // push dword ptr [ebp + param_7]
+        // push dword ptr [ebp + param_6]
+        OV_MATCH(0x0E, 0xFF, 0x75, 0x2C, 0xFF, 0x75, 0x28, 0xFF, 0x75, 0x24, 0xFF, 0x75, 0x20, 0xFF, 0x75, 0x1C),
+OOVPA_END;
+
+// ******************************************************************
+// * XOnlineMatchSessionUpdate
+// ******************************************************************
+OOVPA_XREF(XOnlineMatchSessionUpdate, 4831, 1+23,
+
+    XRefNoSaveIndex,
+    XRefOne)
+
+        XREF_ENTRY(0x28, XREF_CXo_XOnlineMatchSessionUpdate),
+
+        // push ebp
+        // mov ebp, esp
+        // push dword ptr [ebp + param_10]
+        // mov ecx, ...
+        OV_MATCH(0x00, 0x55, 0x8B, 0xEC, 0xFF, 0x75, 0x2C, 0x8B, 0x0D),
+
+        // push dword ptr [ebp + param_9]
+        // push dword ptr [ebp + param_8]
+        // push dword ptr [ebp + param_7]
+        // push dword ptr [ebp + param_6]
+        // push dword ptr [ebp + param_5]
+        OV_MATCH(0x0C, 0xFF, 0x75, 0x28, 0xFF, 0x75, 0x24, 0xFF, 0x75, 0x20, 0xFF, 0x75, 0x1C, 0xFF, 0x75, 0x18),
+OOVPA_END;
+
+// ******************************************************************
+// * CXo::XOnlineMatchSessionCreate
+// ******************************************************************
+OOVPA_XREF(CXo_XOnlineMatchSessionCreate, 4831, 18,
+
+    XREF_CXo_XOnlineMatchSessionCreate,
+    XRefZero)
+
+        // push ebp
+        // mov ebp, esp
+        // push ecx
+        // push ecx
+        // xor eax, eax
+        // cmp ecx, eax
+        // jnz eip + $07h
+        // mov eax, $80150005h
+        // jmp eip + $55h
+        OV_MATCH(0x00, 0x55, 0x8B, 0xEC, 0x51, 0x51, 0x33, 0xC0, 0x3B, 0xC8, 0x75, 0x07, 0xB8, 0x05, 0x00, 0x15, 0x80, 0xEB, 0x55),
```

Comment:
Too much relying on common instruction opcodes than find one or two unique, or uncommon, instruction opcode. Plus 0x80150005 is very common value and should not be rely on it.

Reply:
I'm aware that 0x80150005 is very common, that's why it was only a small part of this signature. I've reworked the signatures anyway, and it's now more relying on relative jump offsets. It worked fine for me, but I have a very limited selection of games to test with.